pkg/communication/client_websocket_transport.go

```diff
@@ -0,0 +1,151 @@
+package communication
+
+import (
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+
+	"golang.org/x/net/websocket"
+
+	"github.com/golang/glog"
+)
+
+// Implementation of the ITransport for websocket communication to send and receive serialized protobuf message bytes
+type ClientWebsocketTransport struct {
+	//VmtServerAddress string
+	//LocalAddress     string
+	//ServerUsername   string
+	//ServerPassword   string
+	ws            *websocket.Conn // created during registration
+	inputStream   chan []byte
+	closeReceived bool
+}
+
+// Instantiate a new ClientWebsocketTransport endpoint for the client
+// Websocket connection is established with the server
+func CreateClientWebsocketTransport(connConfig *WebsocketConnectionConfig) *ClientWebsocketTransport {
+	transport := &ClientWebsocketTransport{
+		ws:          newWebsocketConnection(connConfig),
+		inputStream: make(chan []byte),
+	}
+
+	// Message handler for received messages
+	go transport.waitForServerMessage()
+	return transport
+}
+
+// Create the websocket connection and establish session with the server
+func newWebsocketConnection(connConfig *WebsocketConnectionConfig) *websocket.Conn {
+	// Websocket URL
+	vmtServerUrl := CreateWebsocketTransportPointUrl(connConfig)
+	//
+	localAddr := "http://127.0.0.1" //TODO: required in url format, but ip is don't care for us
+	glog.V(3).Infof("Dial Server: %s", vmtServerUrl)
+
+	config, err := websocket.NewConfig(vmtServerUrl, localAddr)
+	if err != nil {
+		glog.Fatal(err)
+	}
+	usrpasswd := []byte(connConfig.ServerUsername + ":" + connConfig.ServerPassword)
+
+	config.Header = http.Header{
+		"Authorization": {"Basic " + base64.StdEncoding.EncodeToString(usrpasswd)},
+	}
+	config.TlsConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
By default, we are redirecting every traffic to https now right, so it's supposed to be here I think